crates/ruma-common/src/identifiers/key_id.rs

```diff
@@ -12,6 +12,33 @@ use super::{
 };
 
 /// A key algorithm and key name delimited by a colon.
+///
+/// Examples of the use of this struct are [`CrossSigningKeyId`] and [`DeviceKeyId`], which provide
+/// [signing key identifiers](https://spec.matrix.org/v1.12/appendices/#signing-details) for
+/// WHAT??? and devices??? respectively.
```

Comment:
`DeviceKeyId` is not actually just used for **signing key** identifiers.

Each device has two identity keys: an ed25519 key, which is used for signing things; but also a curve25519 key, which is used (via DH) for calculating shared secrets.

So maybe:

```suggestion
/// Examples of the use of this struct are [`DeviceKeyId`], which identifies a Ed25519 or Curve25519 
/// [device key](https://spec.matrix.org/v1.12/client-server-api/#device-keys), and
/// [`CrossSigningKeyId`], which identifies a user's [cross signing key](https://spec.matrix.org/v1.12/client-server-api/#cross-signing).
///
/// This format of identifier is often used in the `signatures` field of [signed JSON](https://spec.matrix.org/v1.12/appendices/#signing-details)
/// where it is referred to as a "signing key identifier".
```

Reply:
Great, using that. Thank you.